### ISSUE TITLE
Reloading save states causes VGA screen corruption

### ISSUE THREAD

**Issue by Dazdfa** [joncampbell123/dosbox-x#2674]
I've tried this on Windows with the 32- and 64-bit Visual Studio and mingw builds of DosBox-X 0.83.15.

Whenever I save a state from a VGA game, if I restart DosBox and try to reload it, I get a corrupted screen display. It appears that some parts of the video state aren't being restored properly.

If I rerun the game from scratch and _then_ load the state, the screen seems fine.

See attached files with the game 7 Colors as an example.

This is a normal game screen.
![normal-game](https://user-images.githubusercontent.com/2503354/125494126-f370b92d-3e39-47f3-b746-2cf9480d99ba.png)

This is what I get if I save from there, quit, restart DosBox, then load the state. The screen is corrupted, although the game is still running.
![reload-after-restart](https://user-images.githubusercontent.com/2503354/125494149-f2a4ca36-6d74-4d73-bd7d-6d9fa87d9e46.png)

This is what I get if I load 7 Colors again then reload the state while it's running. As you can see, it looks intact.
![reload-after-game](https://user-images.githubusercontent.com/2503354/125494214-38f6ef9f-f2b8-488b-90f9-90b3daf84118.png)

**Comment by Wengier** [comment]
Indeed, the save state feature is not perfect. It works best if you load the saved game in the same session, rather than restarting DOSBox-X and then load it. It can be further improved of course, but this apparently requires some work.

**Comment by clst** [comment]
This is what happens to me when loading a save state in Jill of the Jungle even when I am in game.  
Game runs fine but the graphics seem to switch to text mode when loading a savestate.
